### PR TITLE
DM-15151: Set symbol visibility to hidden in pybind11 wrappers

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: include/lsst/daf/base/PropertySet.h [hunk 1]
```diff
@@ -172,6 +173,15 @@ class PropertySet : public Persistable, public Citizen {
      */
     std::type_info const& typeOf(std::string const& name) const;
 
+    /**
+     * Get type info for the specified class
+     *
+     * Implemented in the .cc file to work around symbol visiblity issues on macOS
+     * e.g. https://github.com/pybind/pybind11/issues/1503
+     */
```
Comment (kfindeisen) [inline include/lsst/daf/base/PropertySet.h]: The location of the implementation doesn't seem like something we want in the API documentation. Consider moving to a normal comment.

Comment (kfindeisen) [reply include/lsst/daf/base/PropertySet.h]: Should this be a method on `PropertySet` in the first place? If it's only needed for pybind11, maybe it should be a separate function (in the same files as now, I take it?).

File: python/lsst/daf/base/propertyContainer/propertySet.cc [hunk 2]
```diff
@@ -54,7 +54,9 @@ PYBIND11_MODULE(propertySet, mod) {
             .def("__eq__",
                  [](std::type_info const& self, std::type_info const& other) { return self == other; })
             .def("__ne__",
-                 [](std::type_info const& self, std::type_info const& other) { return self != other; });
+                 [](std::type_info const& self, std::type_info const& other) { return self != other; })
+            .def("name", &std::type_info::name)
+            .def("hash_code", &std::type_info::hash_code);
```
Comment (kfindeisen) [inline python/lsst/daf/base/propertyContainer/propertySet.cc]: Not `__hash__`?

File: include/lsst/daf/base/PropertySet.h (hunk 1 above)
Comment (r-owen) [reply include/lsst/daf/base/PropertySet.h]: It is also used by C++ code. I'll change the "Implemented" comment as per your request.

File: python/lsst/daf/base/propertyContainer/propertySet.cc (hunk 2 above)
Comment (r-owen) [reply python/lsst/daf/base/propertyContainer/propertySet.cc]: I fear that will make it harder to find (I only added these fields for debugging). But what the heck, I'll change it. If folks really want it they can read the wrapper.

**Comment by r-owen** [comment]
Having ripped out boost persistence in DM-15599 the symbol export is no longer necessary. I will remove it assuming a full Jenkins run (including ci_hsc) passes.

**Pull request merged**